From: Martin Klapetek <mklapetek@kde.org>
Date: Fri, 08 Jan 2016 21:41:10 +0000
Subject: Duplicate the file descriptor while opening the file to read the env
X-Git-Url: http://quickgit.kde.org/?p=kwallet.git&a=commitdiff&h=901c27ca7cb05cca1a960747b280d40cd7707158
---
Duplicate the file descriptor while opening the file to read the env

Closing the fd results in kwallet being in a blocked state
with all requests to it being timed out.

This fixes the issue.

Thanks to Christoph Cullmann suggesting the fix.

REVIEW: 126681
---


--- a/src/runtime/kwalletd/main.cpp
+++ b/src/runtime/kwalletd/main.cpp
@@ -93,7 +93,7 @@
     char str[BSIZE] = {'\0'};
 
     int chop = 0;
-    FILE *s3 = fdopen(s2, "r");
+    FILE *s3 = fdopen(dup(s2), "r");
     while(!feof(s3)) {
         if (fgets(str, BSIZE, s3)) {
             chop = strlen(str) - 1;


